lionwebrepo-client/src/main/kotlin/com/strumenta/kolasu/lionwebclient/Utils.kt

```diff
@@ -0,0 +1,47 @@
+package com.strumenta.kolasu.lionwebclient
+
+import com.strumenta.kolasu.lionweb.KNode
+import com.strumenta.kolasu.lionweb.LWNode
+import com.strumenta.kolasu.lionweb.StarLasuLWLanguage
+import com.strumenta.kolasu.lionweb.children
+import com.strumenta.kolasu.model.Node
+import com.strumenta.kolasu.model.Source
+import io.lionweb.lioncore.java.language.Concept
+import io.lionweb.lioncore.java.model.impl.DynamicNode
+import io.lionweb.lioncore.java.model.impl.ProxyNode
+
+fun Node.withSource(source: Source): Node {
+    this.setSourceForTree(source)
+    require(this.source === source)
+    return this
+}
+
+fun LWNode.setParentID(parentID: String?) {
+    val parent =
+        if (parentID == null) {
+            null
+        } else {
+            ProxyNode(parentID)
+        }
+    (this as DynamicNode).parent = parent
```

Comment:
Shouldn't this method be defined on `DynamicNode` then?

Reply:
You are right. I will change it to work on `HasSettableParent` and not on `LWNode`